memsource/api.py

```
@@ -588,6 +590,21 @@ def insert(self, translation_memory_id, target_lang, source_segment, target_segm
 
         self._post('transMemory/insert', params)
 
+    def deleteSourceAndTranslations(
+            self, translation_memory_id: int, segment_ids: List[str]) -> None:
+        """Delete segments from a translation memory.
+
+        :param translation_memory_id: Delete the segments from this translation
+        memory.
+        :param segment_ids: Delete these segments from the translation memory.
+        You cannot pass more than 1000 ids one time.
+        """
+        assert 1000 < len(segment_ids), "You cannot pass more than 1000 ids one time."
```

According to the article (and what I know of other languages), if we compile without support for `asserts` (which, in other languages is the default for "production" code), users of the library will not get an AssertionError when passing more than 1000 ids.

You can return a custom exception here, instead. Won't get a `400`.